rules/falco_rules.yaml

```diff
@@ -3150,10 +3153,15 @@
 - list: white_listed_modules
   items: []
 
+# init_module syscall is available since Falco 0.35.0
+# rule coverage now extends to modprobe usage
 - rule: Linux Kernel Module Injection Detected
   desc: Detect kernel module was injected (from container).
-  condition: spawned_process and container and proc.name=insmod and not proc.args in (white_listed_modules) and thread.cap_effective icontains sys_module
-  output: Linux Kernel Module injection using insmod detected (user=%user.name user_loginuid=%user.loginuid parent_process=%proc.pname module=%proc.args %container.info image=%container.image.repository:%container.image.tag)
+  condition: ((spawned_process and proc.name=insmod) or init_module)
+    and container
+    and not proc.args in (white_listed_modules)
```

Comment:
Nice, thanks for noticing!
Btw, I am thinking that maybe it's time to change this rule and make it even better.
I would like to drop the part of the condition relying on `insmod`, and use instead the related syscalls, which is of course `init_module` but also `finit_module` (which is what my `insmod` is using). wdyt? :)

Reply:
Thank you @loresuso this is great! Done :)

Reply:
Nice! One last thing is either throw away the list `white_listed_modules` cause now it should be unused.
I'd add an `allowed_container_images_loading_kernel_module` list and add it to the condition with `and not` and it's perfect for me!